src/Reducer.js

```diff
@@ -163,7 +163,18 @@ function inject(state, action, props, scenes) {
     case ActionConst.JUMP:
       assert(state.tabs, `Parent=${state.key} is not tab bar, jump action is not valid`);
       ind = -1;
-      state.children.forEach((c, i) => { if (c.sceneKey === action.key) { ind = i; } });
+      state.children.forEach((child, i) => {
+        if (child.sceneKey === action.key) {
+          ind = i;
+
+          // Pass the props to the wrapped tab scene
+          if (props.children) delete props.children;
+          const key = child.children[0].sceneKey;
+          const wrappedScene = scenes[key];
+          child.children = [{ ...props, ...wrappedScene }];
+        }
+      });
+
       assert(ind !== -1, `Cannot find route with key=${action.key} for parent=${state.key}`);
```

Comment:
You should not modify existing state, but produce new one. Here i see you modify existing child within existing state var.